sqldelight-gradle-plugin/src/main/kotlin/app/cash/sqldelight/gradle/SqlDelightDatabase.kt

```diff
@@ -204,6 +210,26 @@ abstract class SqlDelightDatabase @Inject constructor(
     }.distinct()
   }
 
+  /**
+   * Helper to wire relevant task dependencies for dependent projects' tasks. This is necessary to work in
+   * Gradle 8.0+, where otherwise-implicit task dependencies are not allowed.
+   *
+   * Note that we use [Task.mustRunAfter] as the lightest touch option, as that only enforces ordering if the other
+   * project's task is scheduled to run in the current invocation.
+   */
+  private fun <T : Task> wireDependencyTaskDependencies(
```

Comment:
I understand the changes but not why it is required now and the normald output task relationship from Gradle does not work anymore.

Reply:
It never did, and in gradle 8 that implicit relationship becomes an error


```
~/dev/android/personal/CatchUp   z/sqldelightMore ±✚  gw generateMainGemojiDatabaseSchema --rerun-tasks --no-configuration-cache
Type-safe project accessors is an incubating feature.

> Configure project :app
Using com.bugsnag.android.gradle:7+ with Android Gradle Plugin 8+ is not formally supported, and may lead to compatibility errors. For more information, please see https://docs.bugsnag.com/build-integrations/gradle/

> Task :libraries:gemoji:generator:generateMainGemojiDatabaseSchema FAILED
Kotlin build report is written to file:///Users/zacsweers/dev/android/personal/CatchUp/build/reports/kotlin-build/Catchup-build-2023-02-26-22-50-31.txt

FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':libraries:gemoji:generator:generateMainGemojiDatabaseSchema' (type 'GenerateSchemaTask').
  - Gradle detected a problem with the following location: '/Users/zacsweers/dev/android/personal/CatchUp/libraries/gemoji/db/src/main/sqldelight'.
    
    Reason: Task ':libraries:gemoji:generator:generateMainGemojiDatabaseSchema' uses this output of task ':libraries:gemoji:db:generateMainGemojiDatabaseSchema' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':libraries:gemoji:db:generateMainGemojiDatabaseSchema' as an input of ':libraries:gemoji:generator:generateMainGemojiDatabaseSchema'.
      2. Declare an explicit dependency on ':libraries:gemoji:db:generateMainGemojiDatabaseSchema' from ':libraries:gemoji:generator:generateMainGemojiDatabaseSchema' using Task#dependsOn.
      3. Declare an explicit dependency on ':libraries:gemoji:db:generateMainGemojiDatabaseSchema' from ':libraries:gemoji:generator:generateMainGemojiDatabaseSchema' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.0.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
```

Reply:
Could it be the same issue as https://github.com/cashapp/sqldelight/pull/3833? 
Basically the schema output and the .sq inputs need to be in different directory trees.